compiler/src/dotty/tools/dotc/config/Feature.scala

```diff
@@ -128,7 +128,8 @@ object Feature:
 
   def scala2ExperimentalMacroEnabled(using Context) = enabled(scala2macros)
 
-  def betterMatchTypeExtractorsEnabled(using Context) = enabled(betterMatchTypeExtractors)
+  def betterMatchTypeExtractorsEnabled(using Context) =
+    sourceVersion.isAtLeast(`3.6`) || enabled(betterMatchTypeExtractors)
```

Comment:
I suggest just completely removing the check (and deprecating the experimental definition) since we don't expect the new behavior to break any code.

Reply:
We have already set the precedence that we require the import/flag if the source version is lower than the stabilisation version. See for example: https://github.com/scala/scala3/pull/20861. I think we should stay consistent when taking similar decisions.

Reply:
Sounds fine to me here, but in general we'll have to be careful to not make the code harder to read with too many config switches

Reply:
Agreed

Reply:
As discussed last week, and for quick and easy maintainability of the code base, the `betterMatchTypeExtractors` language import will have no effect starting from 3.6.0 and setting the `-source` flag will not change the behaviour.